nuxeo-drive-client/nxdrive/engine/username_resolver.py

```diff
@@ -0,0 +1,41 @@
+'''
+Created on Jul 13, 2016
+
+@author: mkeshava
+'''
+from nxdrive.engine.workers import PollWorker
+from datetime import timedelta, datetime
+from nxdrive.logging_config import get_logger
+
+
+USERNAME_UPDATE_INTERVAL = timedelta(days=1).total_seconds()
+log = get_logger(__name__)
+
+
+class UserNameResolver(PollWorker):
+    '''
+    A Polling Worker to refresh the userid to user's full name mapping everyday
+    '''
+    def __init__(self, engine):
+        '''
+        The polling interval is 1 day
+        '''
+        super(UserNameResolver, self).__init__(check_interval=USERNAME_UPDATE_INTERVAL)
+        self._engine = engine
```

Comment:
don't need this, if all methods move from the engine class here. Instead, inject in the c-tor what's needed: dao, rest client..